google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/StreamingSubscriberConnection.java

```diff
@@ -211,35 +211,45 @@ private boolean isAlive() {
   @Override
   public void sendAckOperations(
       List<String> acksToSend, List<PendingModifyAckDeadline> ackDeadlineExtensions) {
+    StreamingPullRequest.Builder[] requests =
+        partitionAckOperations(acksToSend, ackDeadlineExtensions, MAX_PER_REQUEST_CHANGES);
+    for (StreamingPullRequest.Builder request : requests) {
+      requestObserver.onNext(request.build());
+    }
+  }
 
-    // Send the modify ack deadlines in batches as not to exceed the max request
-    // size.
-    List<List<String>> ackChunks = Lists.partition(acksToSend, MAX_PER_REQUEST_CHANGES);
-    List<List<PendingModifyAckDeadline>> modifyAckDeadlineChunks =
-        Lists.partition(ackDeadlineExtensions, MAX_PER_REQUEST_CHANGES);
-    Iterator<List<String>> ackChunksIt = ackChunks.iterator();
-    Iterator<List<PendingModifyAckDeadline>> modifyAckDeadlineChunksIt =
-        modifyAckDeadlineChunks.iterator();
-
-    while (ackChunksIt.hasNext() || modifyAckDeadlineChunksIt.hasNext()) {
-      StreamingPullRequest.Builder requestBuilder =
-          StreamingPullRequest.newBuilder();
-      if (modifyAckDeadlineChunksIt.hasNext()) {
-        List<PendingModifyAckDeadline> modAckChunk = modifyAckDeadlineChunksIt.next();
-        for (PendingModifyAckDeadline modifyAckDeadline : modAckChunk) {
-          for (String ackId : modifyAckDeadline.ackIds) {
-            requestBuilder
-                .addModifyDeadlineSeconds(modifyAckDeadline.deadlineExtensionSeconds)
-                .addModifyDeadlineAckIds(ackId);
-          }
-        }
-      }
-      if (ackChunksIt.hasNext()) {
-        List<String> ackChunk = ackChunksIt.next();
-        requestBuilder.addAllAckIds(ackChunk);
+  @VisibleForTesting
```

Comment:
You made this visible for testing but there doesn't appear to be any test changes to access it.